## Final Project Submission

Please fill out:
* Student name: 
* Student pace: self paced / part time / full time
* Scheduled project review date/time: 
* Instructor name: 
* Blog post URL:


In [1]:
# Importing the necessary libraries to be used in the analysis
import numpy as np
import pandas as pd
import statsmodels.api as sm
import scipy.stats as stats
import matplotlib.pyplot as plt
%matplotlib inline

import warnings
warnings.filterwarnings("ignore")

## Data Preparation
Load the dataset to be used in the analysis

In [2]:
#Importing the data set
housing_data = pd.read_csv("data/kc_house_data.csv") 

id        date   price  bedrooms  bathrooms  sqft_living  sqft_lot  \
0  7129300520  10/13/2014  221900         3       1.00         1180      5650   
1  6414100192  12/09/2014  538000         3       2.25         2570      7242   
2  5631500400   2/25/2015  180000         2       1.00          770     10000   
3  2487200875  12/09/2014  604000         4       3.00         1960      5000   
4  1954400510   2/18/2015  510000         3       2.00         1680      8080   

   floors waterfront  view  ...          grade sqft_above  sqft_basement  \
0     1.0        NaN  NONE  ...      7 Average       1180              0   
1     2.0         NO  NONE  ...      7 Average       2170            400   
2     1.0         NO  NONE  ...  6 Low Average        770              0   
3     1.0         NO  NONE  ...      7 Average       1050            910   
4     1.0         NO  NONE  ...         8 Good       1680              0   

  yr_built  yr_renovated  zipcode      lat     long  sqft_living15  sqft_lot15  
0     1955           0.0    98178  47.5112 -122.257           1340        5650  
1     1951        1991.0    98125  47.7210 -122.319           1690        7639  
2     1933           NaN    98028  47.7379 -122.233           2720        8062  
3     1965           0.0    98136  47.5208 -122.393           1360        5000  
4     1987           0.0    98074  47.6168 -122.045           1800        7503  

[5 rows x 21 columns]

In [3]:
# Previewing the datset
housing_data.head()

id        date   price  bedrooms  bathrooms  sqft_living  sqft_lot  \
0  7129300520  10/13/2014  221900         3       1.00         1180      5650   
1  6414100192  12/09/2014  538000         3       2.25         2570      7242   
2  5631500400   2/25/2015  180000         2       1.00          770     10000   
3  2487200875  12/09/2014  604000         4       3.00         1960      5000   
4  1954400510   2/18/2015  510000         3       2.00         1680      8080   

   floors waterfront  view  ...          grade sqft_above  sqft_basement  \
0     1.0        NaN  NONE  ...      7 Average       1180              0   
1     2.0         NO  NONE  ...      7 Average       2170            400   
2     1.0         NO  NONE  ...  6 Low Average        770              0   
3     1.0         NO  NONE  ...      7 Average       1050            910   
4     1.0         NO  NONE  ...         8 Good       1680              0   

  yr_built  yr_renovated  zipcode      lat     long  sqft_living15  sqft_lot15  
0     1955           0.0    98178  47.5112 -122.257           1340        5650  
1     1951        1991.0    98125  47.7210 -122.319           1690        7639  
2     1933           NaN    98028  47.7379 -122.233           2720        8062  
3     1965           0.0    98136  47.5208 -122.393           1360        5000  
4     1987           0.0    98074  47.6168 -122.045           1800        7503  

[5 rows x 21 columns]

In [4]:
housing_data.tail()

id        date   price  bedrooms  bathrooms  sqft_living  \
21592   263000018   5/21/2014  360000         3       2.50         1530   
21593  6600060120   2/23/2015  400000         4       2.50         2310   
21594  1523300141   6/23/2014  402101         2       0.75         1020   
21595   291310100   1/16/2015  400000         3       2.50         1600   
21596  1523300157  10/15/2014  325000         2       0.75         1020   

       sqft_lot  floors waterfront  view  ...      grade sqft_above  \
21592      1131     3.0         NO  NONE  ...     8 Good       1530   
21593      5813     2.0         NO  NONE  ...     8 Good       2310   
21594      1350     2.0         NO  NONE  ...  7 Average       1020   
21595      2388     2.0        NaN  NONE  ...     8 Good       1600   
21596      1076     2.0         NO  NONE  ...  7 Average       1020   

       sqft_basement yr_built  yr_renovated  zipcode      lat     long  \
21592              0     2009           0.0    98103  47.6993 -122.346   
21593              0     2014           0.0    98146  47.5107 -122.362   
21594              0     2009           0.0    98144  47.5944 -122.299   
21595              0     2004           0.0    98027  47.5345 -122.069   
21596              0     2008           0.0    98144  47.5941 -122.299   

       sqft_living15  sqft_lot15  
21592           1530        1509  
21593           1830        7200  
21594           1020        2007  
21595           1410        1287  
21596           1020        1357  

[5 rows x 21 columns]

## Data Understanding
This involves exploring the dataset's characteristics, structure, and key attributes. This involves an exploration of the dataset's properties, structure, and crucial characteristics. The key steps include investigating:

The dataset's shape

6. Existence of duplicate entries -->

In [9]:
# check no.or rows and columns
housing_data.shape

(21597, 21)

The dataset's shape is (21597, 21), this meansthat the data has 21 columns and 21597 rows.

Column names

In [10]:
# check columns names
housing_data.columns

Index(['id', 'date', 'price', 'bedrooms', 'bathrooms', 'sqft_living',
       'sqft_lot', 'floors', 'waterfront', 'view', 'condition', 'grade',
       'sqft_above', 'sqft_basement', 'yr_built', 'yr_renovated', 'zipcode',
       'lat', 'long', 'sqft_living15', 'sqft_lot15'],
      dtype='object')

Data types of columns

In [11]:
housing_data.dtypes

id                 int64
date              object
price              int64
bedrooms           int64
bathrooms        float64
sqft_living        int64
sqft_lot           int64
floors           float64
waterfront        object
view              object
condition         object
grade             object
sqft_above         int64
sqft_basement     object
yr_built           int64
yr_renovated     float64
zipcode            int64
lat              float64
long             float64
sqft_living15      int64
sqft_lot15         int64
dtype: object

Summary statistics of the data

In [17]:
# A statistical summary of the data
housing_data.describe()

id         price      bedrooms     bathrooms   sqft_living  \
count  2.159700e+04  2.159700e+04  21597.000000  21597.000000  21597.000000   
mean   4.580474e+09  5.402966e+05      3.373200      2.115826   2080.321850   
std    2.876736e+09  3.673681e+05      0.926299      0.768984    918.106125   
min    1.000102e+06  7.800000e+04      1.000000      0.500000    370.000000   
25%    2.123049e+09  3.220000e+05      3.000000      1.750000   1430.000000   
50%    3.904930e+09  4.500000e+05      3.000000      2.250000   1910.000000   
75%    7.308900e+09  6.450000e+05      4.000000      2.500000   2550.000000   
max    9.900000e+09  7.700000e+06     33.000000      8.000000  13540.000000   

           sqft_lot        floors    sqft_above      yr_built  yr_renovated  \
count  2.159700e+04  21597.000000  21597.000000  21597.000000  17755.000000   
mean   1.509941e+04      1.494096   1788.596842   1970.999676     83.636778   
std    4.141264e+04      0.539683    827.759761     29.375234    399.946414   
min    5.200000e+02      1.000000    370.000000   1900.000000      0.000000   
25%    5.040000e+03      1.000000   1190.000000   1951.000000      0.000000   
50%    7.618000e+03      1.500000   1560.000000   1975.000000      0.000000   
75%    1.068500e+04      2.000000   2210.000000   1997.000000      0.000000   
max    1.651359e+06      3.500000   9410.000000   2015.000000   2015.000000   

            zipcode           lat          long  sqft_living15     sqft_lot15  
count  21597.000000  21597.000000  21597.000000   21597.000000   21597.000000  
mean   98077.951845     47.560093   -122.213982    1986.620318   12758.283512  
std       53.513072      0.138552      0.140724     685.230472   27274.441950  
min    98001.000000     47.155900   -122.519000     399.000000     651.000000  
25%    98033.000000     47.471100   -122.328000    1490.000000    5100.000000  
50%    98065.000000     47.571800   -122.231000    1840.000000    7620.000000  
75%    98118.000000     47.678000   -122.125000    2360.000000   10083.000000  
max    98199.000000     47.777600   -121.315000    6210.000000  871200.000000

Existence of missing data

In [19]:
# check the overview of the df
housing_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21597 entries, 0 to 21596
Data columns (total 21 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   id             21597 non-null  int64  
 1   date           21597 non-null  object 
 2   price          21597 non-null  int64  
 3   bedrooms       21597 non-null  int64  
 4   bathrooms      21597 non-null  float64
 5   sqft_living    21597 non-null  int64  
 6   sqft_lot       21597 non-null  int64  
 7   floors         21597 non-null  float64
 8   waterfront     19221 non-null  object 
 9   view           21534 non-null  object 
 10  condition      21597 non-null  object 
 11  grade          21597 non-null  object 
 12  sqft_above     21597 non-null  int64  
 13  sqft_basement  21597 non-null  object 
 14  yr_built       21597 non-null  int64  
 15  yr_renovated   17755 non-null  float64
 16  zipcode        21597 non-null  int64  
 17  lat            21597 non-null  float64
 18  long  

"waterfront" has 19221 non-null values, indicating that it has missing values because it's less than the total number of rows (21597).
"view" has 21534 non-null values, suggesting it also contains missing data.
"yr_renovated" has 17755 non-null values, indicating missing values in this column as well.